src/main/java/io/tackle/operator/MicroserviceController.java

```diff
@@ -13,32 +15,49 @@
 
 import javax.inject.Inject;
 
+import static io.tackle.operator.Utils.metadataName;
+
 @Controller(namespaces = Controller.WATCH_CURRENT_NAMESPACE)
-public class MicroserviceController extends AbstractController implements ResourceController<Microservice> {
+public class MicroserviceController implements ResourceController<Microservice> {
 
     private final Logger log = Logger.getLogger(getClass());
     @Inject
     KubernetesClient kubernetesClient;
+    @Inject
+    PostgreSQLDeployer postgreSQLDeployer;
+    @Inject
+    RestDeployer restDeployer;
 
     @Override
     public UpdateControl<Microservice> createOrUpdateResource(Microservice microservice, Context<Microservice> context) {
         String namespace = microservice.getMetadata().getNamespace();
         String name = metadataName(microservice);
-        log.infof("Execution createOrUpdateResource for '%s' in namespace '%s'", name, namespace);
 
-        MixedOperation<PostgreSQL, KubernetesResourceList<PostgreSQL>, Resource<PostgreSQL>> postgreSQLClient = kubernetesClient.customResources(PostgreSQL.class);
-        final String postgresqlCR = microservice.getSpec().getPostgreSQL();
-        PostgreSQL postgreSQL = postgreSQLClient.load(TackleController.class.getResourceAsStream(postgresqlCR)).get();
+        MixedOperation<Microservice, KubernetesResourceList<Microservice>, Resource<Microservice>> microserviceClient = kubernetesClient.customResources(Microservice.class);
+        final String restImage = microservice.getSpec().getRestImage();
+        ListOptions listOptions = new ListOptionsBuilder().withFieldSelector("metadata.name!=" + name).build();
+        MixedOperation<Tackle, KubernetesResourceList<Tackle>, Resource<Tackle>> tackleClient = kubernetesClient.customResources(Tackle.class);
+        if (tackleClient.inNamespace(namespace).list().getItems().isEmpty()) {
+            log.errorf("Standalone '%s' Microservice CR isn't allowed: create a Tackle CR to instantiate Tackle application", name);
+            microserviceClient.delete(microservice);
+            return UpdateControl.noUpdate();
+        } else if (microserviceClient
+                .inNamespace(namespace)
+                .list(listOptions)
+                .getItems()
+                .stream()
+                .anyMatch(microserviceAlreadyDeployed -> restImage.equals(microserviceAlreadyDeployed.getSpec().getRestImage()))) {
```

Comment:
This is considering that the user can not have more than 1 tackle per namespace. Shouldn't this be constraining per Tackle CR name ?

Reply:
:+1: 
Implemented the "Multiple Tackle CRs in a single namespace" approach.